external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpoutRetryExponentialBackoff.java

```diff
@@ -290,8 +291,8 @@ public int readyMessageCount() {
     }
 
     @Override
-    public KafkaSpoutMessageId getMessageId(ConsumerRecord<?, ?> record) {
-        KafkaSpoutMessageId msgId = new KafkaSpoutMessageId(record);
+    public KafkaSpoutMessageId getMessageId(ConsumerRecord<?, ?> record, TopologyContext context) {
```

Comment:
I'm fine with this in 2.x, but for 1.2.0 I think we should try to avoid breaking changes (RetryService is a public API, it's the same reason this method still takes a ConsumerRecord instead of just the partition and offset).

Reply:
I guess the only non-backwards compatible changes are in case the user implemented their own version of KafkaSpoutRetryService. Right? I will check if it is possible to make this fix without this change. If possible I will do.

Reply:
I think if we split KafkaSpoutMessageId we won't need the context here, so hopefully it won't be a problem to fix.

Reply:
Done